SBTi/data/cdp.py

```diff
@@ -0,0 +1,78 @@
+import pandas as pd
+from SBTi.data.data_provider import DataProvider
+
+
+class CDP(DataProvider):
+    """
+    Data provider skeleton for CDP.
+    """
+
+    def get_targets(self, companies: list) -> pd.DataFrame:
+        """
+        Get all the targets for the whole portfolio of companies. This should return a dataframe, containing at least
+        the following columns:
+
+        * company_name: The name of the company
+        * company_id: The ID of the company
+        * target_reference_number: Int *x* of Abs *x*
+        * target_reference_number: Int *x* of Abs *x*
+        * scope: The scope of the target. This should be a valid scope in the SR15 mapping
+        * base_year: The base year of the target
+        * start_year: The start year of the target
+        * target_year: The year when the target should be achieved
+        * reduction_from_base_year: Targeted reduction in emissions from the base year
+        * emissions_in_scope: Company emissions in the target's scope at start of the base year
+        * achieved_reduction: The emission reduction that has already been achieved
+
+        :param companies: A list of companies. Each company should be a dict with a "company_name" and "company_id"
+                            field.
+        :return: A dataframe containing the targets
+        """
+        # TODO: Make an API request
+        # TODO: Transform the result into a dataframe
+        # TODO: Make sure the columns align with those defined in the docstring
+        raise NotImplementedError
+
+    def get_company_data(self, companies: list) -> pd.DataFrame:
```

Comment:
maybe doesn't need to be part of this PR, but it might be better if the output of this function isn't a DataFrame, but instead an array of a typed object that clearly defines the contract that is needed for company data. Otherwise you'll always have to work with an untyped dataframe and it's much easier to map it wrong/make errors